deps/build.jl

```diff
@@ -1,133 +1,61 @@
-    "libswscale2-ffmpeg"    => libswscale,
-    # ffmpeg is again available in standard packages as of ubuntu 15.04
-    "libavcodec-ffmpeg56"   => libavcodec,
-    "libavdevice-ffmpeg56"  => libavdevice,
-    "libavfilter-ffmpeg5"   => libavfilter,
-    "libavformat-ffmpeg56"  => libavformat,
-    "libavutil-ffmpeg54"    => libavutil,
-    "libswscale-ffmpeg3"    => libswscale,
-    #"libavresample-ffmpeg2" => libavresample,
-)
-
-#if have_swresample
-#    apt_packages["libswresample0-ffmpeg"] = libswresample
-#end
-
-#if have_avresample
-#    apt_packages["libavresample1"] = libavresample
-#end
-
-provides(AptGet,
-         apt_packages)
-
-provides(Yum,
-         Dict("ffmpeg" => libav_libs))
-
-provides(Pacman,
-         Dict("ffmpeg" => libav_libs))
-
-provides(Sources,
-         URI("http://www.ffmpeg.org/releases/ffmpeg-4.1.tar.gz"),
-         libav_libs)
-
-provides(BuildProcess, Autotools(configure_options=["--enable-gpl"]), libav_libs, os = :Unix)
-
-@BinDeps.install Dict(
-    :libavcodec => :libavcodec,
-    :libavformat => :libavformat,
-    :libavutil => :libavutil,
-    :libswscale => :libswscale,
-    :libavfilter => :libavfilter,
-    :libavdevice => :libavdevice
+using BinaryProvider
+
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
+const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
+
+libpath = joinpath(@__DIR__, "usr/bin") #for forcing LibraryProduct to find libs in bin dir (as ffmpeg likes to do)
+# These are the two binary objects we care about
+products = Product[
+    ExecutableProduct(prefix, "ffmpeg", :ffmpeg),
+    ExecutableProduct(prefix, "ffprobe", :ffprobe),
+    ExecutableProduct(prefix, "ffplay", :ffplay),
+
+    LibraryProduct(libpath, [["libavcodec-ffmpeg.so.","libavcodec.","libavcodec.so.","libavcodec.ffmpeg.so.","avcodec-"].*["53" "54" "55" "56" "57" "58"].*[".dylib"]...], :libavcodec),
+    LibraryProduct(libpath, [["libavformat-ffmpeg.so.","libavformat.","libavformat.so.","libavformat.ffmpeg.so.","avformat-"].*["53" "54" "55" "56" "57" "58"]...], :libavformat),
+    LibraryProduct(libpath, [["libavutil-ffmpeg.so.", "libavutil.","libavutil.so.", "libavutil.ffmpeg.so.", "avutil-"].*["51" "52" "54" "55" "56"]...], :libavutil),
+    LibraryProduct(libpath, [["libswscale-ffmpeg.so.","libswscale.","libswscale.so.","libswscale.ffmpeg.so.","swscale-"].*["2" "3" "4" "5"]...], :libswscale),
+    LibraryProduct(libpath, [["libavfilter-ffmpeg.so.","libavfilter.","libavfilter.so.","libavfilter.ffmpeg.so.","avfilter-"].*["2" "3" "4" "5" "6" "7"]...], :libavfilter),
+    LibraryProduct(libpath, [["libavdevice-ffmpeg.so.","libavdevice.","libavdevice.so.","libavdevice.ffmpeg.so.","avdevice-"].*["53" "54" "55" "56" "57" "58"]...], :libavdevice),
+]
+
+# Download binaries from hosted location
+v = "4.1"
+#bin_prefix = "https://github.com/ianshmean/FFMPEG-tarballs/raw/master/$v/bin"
+bin_prefix = "/Users/IanB/Github_Personal/FFMPEG-tarballs/$v/bin"
```

Comment:
You probably meant to keep this as an http URL.  ;)

Reply:
Indeed! I was working on this on a spotty internet connection so had gone local and forgot to revert. just fixed it. thanks